cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterRebalanceAssemblyOperator.java

```diff
@@ -0,0 +1,616 @@
+        Promise<KafkaClusterRebalanceStatus> p = Promise.promise();
+        if (rebalanceAnnotation == RebalanceAnnotation.none) {
+            log.info("Arming Cruise Control rebalance user task status timer");
+            String sessionId = clusterRebalance.getStatus().getSessionId();
+            vertx.setPeriodic(REBALANCE_TASK_STATUS_TIMER_MS, t -> {
+                resourceOperator.getAsync(clusterRebalance.getMetadata().getNamespace(), clusterRebalance.getMetadata().getName()).setHandler(getResult -> {
+                    if (getResult.succeeded()) {
+                        KafkaClusterRebalance fetchedClusterRebalance = getResult.result();
+                        if (getRebalanceAnnotation(fetchedClusterRebalance) == RebalanceAnnotation.stop) {
+                            log.debug("Stopping current Cruise Control rebalance user task");
+                            vertx.cancelTimer(t);
+                            apiClient.stopExecution(host, 9090 /* TODO: need public constant from CruiseControl model */).setHandler(stopResult -> {
+                                p.complete(new KafkaClusterRebalanceStatusBuilder()
+                                        .withSessionId(null)
+                                        .addNewCondition().withNewType(State.Stopped.toString()).endCondition().build());
+                            });
+                        } else {
+                            log.info("Getting Cruise Control rebalance user task status");
+                            apiClient.getUserTaskStatus(host, 9090 /* TODO: need public constant from CruiseControl model */, sessionId).setHandler(userTaskResult -> {
+                                if (userTaskResult.succeeded()) {
+                                    CruiseControlResponse response = userTaskResult.result();
+                                    JsonObject taskStatusJson = response.getJson();
+                                    String taskStatus = taskStatusJson.getString("Status");
+                                    if (taskStatus.equals("completed")) {
+                                        vertx.cancelTimer(t);
+                                        p.complete(new KafkaClusterRebalanceStatusBuilder()
+                                                .withSessionId(null)
+                                                .withOptimizationResult(taskStatusJson.getJsonObject("rebalance").getMap())
+                                                .addNewCondition().withType(State.Ready.toString()).endCondition().build());
+                                    } else if (taskStatus.equals("aborted")) {
+                                        log.error("Unexpected state {}", taskStatus);
+                                        vertx.cancelTimer(t);
+                                        p.fail("Unexpected state " + taskStatus);
+                                    }
```

Comment:
Again, I would have an `else` to avoid the possibility of an infinite loop if CC returned something unexpected.